File: go/problem014_longest_common_prefix.go
// https://leetcode.com/problems/longest-common-prefix/

func longestCommonPrefix(strs []string) string {
    if len(strs) == 0 {
        return ""
    }

    res := strs[0]

    for _, word := range strs[1:] {

        size := min(len(word), len(res))

        i:=0
        for i<size && res[i] == word[i] {
            i++
        } 
        res = res[:i]
    }

    return res
}

File: python/problem_014_longest_common_prefix.py
# https://leetcode.com/problems/longest-common-prefix

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""

        result = strs[0]

        for word in strs[1:]:
            size = min(len(word), len(result))

            i=0
            while i<size and word[i] == result[i]:
                i += 1
            
            result = result[:i]

        return result

        
